### ISSUE TITLE
[BUG] `df.groupby(['a']).agg({'b': ['mean', 'mean']})` only produces a single column (as opposed to 2)

### ISSUE THREAD

**Issue by MarcoGorelli** [rapidsai/cudf#17649]
**Describe the bug**

**Steps/Code to reproduce bug**

```
df = cudf.DataFrame({'a': [1,1,2], 'b': [4,5,6]})
df.groupby(['a']).agg({'b': ['mean', 'mean']})
```
outputs
```
 	b
	mean
a 	
1 	4.5
2 	6.0
```

**Expected behavior**

what pandas does:
```python
 	b
	mean 	mean
a 		
1 	4.5 	4.5
2 	6.0 	6.0
```

Spotted in [Narwhals](https://github.com/narwhals-dev/narwhals)

**Environment overview (please complete the following information)**
 - Environment location: [Bare-metal, Docker, Cloud(specify cloud provider)]
 - Method of cuDF install: [conda, Docker, or from source]
   - If method of install is [Docker], provide `docker pull` & `docker run` commands used

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

**Additional context**

**Comment by galipremsagar** [comment]
To support this case we will need to add support for duplicate column names in `cudf`, which I don't will be supported in near future. @mroeschke does pandas intend to continue supporting duplicate column name use-case?

**Comment by MarcoGorelli** [comment]
thanks for your response

tbf now that you've added support for `NamedAgg` in cuDF, we can probably rewrite how we do `group_by` in Narwhals for pandas-like libraries and avoid the duplicate column names completely

> @mroeschke does pandas intend to continue supporting duplicate column name use-case?

the last I remember hearing about this was that duplicate column names are a fact of life and should continue to be supported (though to be honest I think they create more problems than they solve)

**Comment by mroeschke** [comment]
Yeah agreed with Marco that I think pandas will continue to support duplicate column names in the foreseeable future

**Comment by vyasr** [comment]
Yes that is consistent with what I've heard in previous discussions about this as well.

**Comment by MarcoGorelli** [comment]
Same for cudf.pandas for this issue

OK with just raising here on cudf's side?
